Landing page hero section

JorgeAMendoza opened the pull request:
Implement homepage style for mobile, tablet, and navigation. There was no Figma design for the navbar, so I created my own design and implementation.

JorgeAMendoza committed:
landing page structure

set HTML structure for the landing page, applying semantic tags. Import
assets for the landing page and placed them within the `public` folder.

JorgeAMendoza committed:
insert missing aria label from section

JorgeAMendoza committed:
mobile nav

begins styling the mobile nav, create button that displays the nav menu
and closes it.

JorgeAMendoza committed:
create mobile nav menu

modify and style the navigation bar to have a mobile friendly design on
mobile devices. Button created to control the nav.

JorgeAMendoza committed:
lock page scroll when mobile nav is open

JorgeAMendoza committed:
style mobile hero

implement styles for the mobile view of the landing page.

JorgeAMendoza committed:
company list

style the company list section for mobile view, replace logo icons with
updated svg's.

JorgeAMendoza committed:
style testimonials

style the testimonial section of the homepage for mobile view.

JorgeAMendoza committed:
add grey border to the mobile nav menu

JorgeAMendoza committed:
navbar tablet

implement design for the navbar for tablet view. Adjust the logic of
locking the scroll on the navbar when the navbar is open. Create new
hook that indicates a mobile or tablet viewport.

JorgeAMendoza committed:
style the application for tablet view

JorgeAMendoza committed:
desktop style

apply desktop styles for the navbar and the hero banner.

JorgeAMendoza committed:
style the companies seciton for dekstop.

JorgeAMendoza committed:
adjust media hook behavior

on first render, the hook checks if its already in a mobile view, if so
it sets the state accordingly.

JorgeAMendoza committed:
testimonials

style the testimonials section for desktop view.

JorgeAMendoza committed:
adjust image width for smaller viewports

JorgeAMendoza committed:
adjust position of the contact button

mentor-tara reviewed (approved):
Go go go :smile:[](prolog-14)

romanchukenator commented on hooks/useIsMobile.tsx:
```diff
@@ -0,0 +1,23 @@
+const useIsMobile = () => {
+  const [isMobile, setIsMobile] = useState(false);
+
+  useEffect(() => {
+    if (window.innerWidth < 768) setIsMobile(true);
+  }, []);
+
+  useEffect(() => {
+    function detectWindowSize() {
+      window.innerWidth < 768 ? setIsMobile(true) : setIsMobile(false);
+    }
+    window.addEventListener("resize", detectWindowSize);
+    return () => {
+      window.removeEventListener("resize", detectWindowSize);
+    };
+  }, []);
+
+  return isMobile;
+};
```
Maybe I'm missing something as I'm not a React-head but couldn't you just define one `useEffect()` function? 

And if you're looking to set a default value to `isMobile` could you call `setIsMobile(window.innerWidth < 768)` outside of the `useEffect` function?

JorgeAMendoza replied:
It appears that we can only access the `window` object when the component mounted, meaning we would only be able to access the `window` object within our side-effects, meaning the `useEffect` Hook.

JorgeAMendoza replied:
I went ahead and just set the `window.innerWidth < 768` in the `useEffect` that initializes the window resize event. Since its going to only be run once, its essentially doing the same thing. This removes the unnecessary `useEffect`

romanchukenator commented on pages/index.jsx:
```diff
@@ -1,20 +1,93 @@
+          }}
+        >
+          {/* eslint-disable-next-line @next/next/no-img-element */}
+          <img src="/icons/nav-menu.svg" alt="menu" />
```
Since you imported the `Image` component why not use it here?

jkettmann replied:
The Image component has a few advantages that are mostly about image optimization. The disadvantage is that you have to define the dimension. Since the image is an SVG there's no need for optimization. At least that's the thought behind using imgs in e.g. the sidebar navigation component

romanchukenator commented on pages/index.jsx:
```diff
@@ -25,10 +98,94 @@ const IssuesPage = () => {
+
+                <div className={styles.customerInfo}>
+                  {/* eslint-disable-next-line @next/next/no-img-element */}
+                  <img
```
Same comment as https://github.com/profydev/prolog-app-JorgeAMendoza/pull/14/files#r1382435102

romanchukenator commented on pages/index.jsx:
```diff
@@ -25,10 +98,94 @@ const IssuesPage = () => {
+              {companyList.map((company) => (
+                <li key={company.name}>
+                  {/* eslint-disable-next-line @next/next/no-img-element */}
+                  <img
```
Same comment as https://github.com/profydev/prolog-app-JorgeAMendoza/pull/14/files#r1382435102

romanchukenator commented on pages/index.jsx:
```diff
@@ -25,10 +98,94 @@ const IssuesPage = () => {
+          </div>
+        </section>
+
+        <section
```
You might maybe want to make this a `<Testimonials />` component? I think it'd make your `index' .jsx and `.scss` files a little cleaner, too, and you'd be able to write some shorter Story code to test individual components. The component could also, like, define a default `const companyList = [...]` but also take in a `companyList` param in case you'd ever want to reuse this list elsewhere.

I mean, this is obviously fine and definitely not wrong and this comment comes mostly from how much trouble I have maintaining concentration over long chunks of code. My eyes glaze-th over! Woe!

Oh, and I'd make this comment about a `<CompanyList />` component.

jkettmann replied:
I agree with splitting up this file into multiple components. There's at least a header, and a testimonial. I wouldn't add a default list of companies though. Adding code for potential future code is usually discouraged

JorgeAMendoza replied:
Implemented the testimonial component in commit 9e80b4

romanchukenator commented on pages/index.module.scss:
```diff
@@ -2,46 +2,215 @@
+  padding-bottom: 4.2em;
+}
+
+.heroContainer {
```
Maybe nest some of these styles? Especially since they're not separate components. This obviously works and there're no styles clashing but reading all the flattened css code makes me irrationally worry. << Real nit-picky comment 🌵 !

romanchukenator reviewed (approved):
Left a few comments and questions but the PR's approved! 👍

JorgeAMendoza committed:
Testimonial Component

For the `index.jsx` page, created a `Testimonials` component that takes
in a list of customers and renders testimonial cards.

JorgeAMendoza committed:
useMobile Hook

set the `window.innerwidth` conditional into the `useEffect` that sets
the document event, essentially doing the same thing and removing the
unnecessary `useEffect`.
